TezosKit/Client/TezosNodeClient.swift

```diff
@@ -538,70 +537,73 @@ public class TezosNodeClient: AbstractClient {
+      let fetchersGroup = DispatchGroup()
+
+      // Fetch data about the chain being operated on.
+      var chainID: String?
+      var headHash: String?
+      var protocolHash: String?
+      let chainHeadRequestRPC = GetChainHeadRPC { json, _ in
+        if let json = json,
+           let fetchedChainID = json["chain_id"] as? String,
+           let fetchedHeadHash = json["hash"] as? String,
+           let fetchedProtocolHash = json["protocol"] as? String {
+          chainID = fetchedChainID
+          headHash = fetchedHeadHash
+          protocolHash = fetchedProtocolHash
+        }
+        fetchersGroup.leave()
       }
-      fetchersGroup.leave()
-    }
 
-    // Fetch data about the address being operated on.
-    var operationCounter: Int?
-    let getAddressCounterRPC =
-      GetAddressCounterRPC(address: address) { fetchedOperationCounter, _ in
+      // Fetch data about the address being operated on.
+      var operationCounter: Int?
+      let getAddressCounterRPC = GetAddressCounterRPC(address: address) { fetchedOperationCounter, _ in
         if let fetchedOperationCounter = fetchedOperationCounter {
           operationCounter = fetchedOperationCounter
         }
         fetchersGroup.leave()
       }
 
-    // Fetch data about the key.
-    var addressKey: String?
-    let getAddressManagerKeyRPC = GetAddressManagerKeyRPC(address: address) { fetchedManagerAndKey, _ in
-      if let fetchedManagerAndKey = fetchedManagerAndKey,
-        let fetchedKey = fetchedManagerAndKey["key"] as? String {
-        addressKey = fetchedKey
+      // Fetch data about the key.
+      var addressKey: String?
+      let getAddressManagerKeyRPC = GetAddressManagerKeyRPC(address: address) { fetchedManagerAndKey, _ in
+        if let fetchedManagerAndKey = fetchedManagerAndKey,
+          let fetchedKey = fetchedManagerAndKey["key"] as? String {
+          addressKey = fetchedKey
+        }
+        fetchersGroup.leave()
       }
-      fetchersGroup.leave()
-    }
 
-    // Send RPCs and wait for results
-    fetchersGroup.enter()
-    send(rpc: chainHeadRequestRPC)
-
-    fetchersGroup.enter()
-    send(rpc: getAddressCounterRPC)
-
-    fetchersGroup.enter()
-    send(rpc: getAddressManagerKeyRPC)
-
-    fetchersGroup.wait()
-
-    // Return fetched data as an OperationData if all data was successfully retrieved.
-    if let operationCounter = operationCounter,
-      let headHash = headHash,
-      let chainID = chainID,
-      let protocolHash = protocolHash {
-      return OperationMetadata(
-        chainID: chainID,
-        headHash: headHash,
-        protocolHash: protocolHash,
-        addressCounter: operationCounter,
-        key: addressKey
-      )
+      // Send RPCs and wait for results
+      fetchersGroup.enter()
+      send(rpc: chainHeadRequestRPC)
+
+      fetchersGroup.enter()
+      send(rpc: getAddressCounterRPC)
+
+      fetchersGroup.enter()
+      send(rpc: getAddressManagerKeyRPC)
+
+      fetchersGroup.wait()
```

Comment:
Update this to use all promises and not need any dispatch groups

Reply:
Acknowledged.